packages/daemon/src/daemon.js

```diff
@@ -107,6 +107,51 @@ const makeEndoBootstrap = (
     return `readable-blob-sha512:${sha512Hex}`;
   };
 
+  /** @type {import('./types.js').IdentifyFromFn} */
+  const identifyFrom = async (originFormulaIdentifier, namePath) => {
+    await null;
+    if (namePath.length === 0) {
+      return originFormulaIdentifier;
+    }
+    // Attempt to shortcut the identify by reading directly from the petStore.
+    const { type: formulaType, number: formulaNumber } = parseFormulaIdentifier(
+      originFormulaIdentifier,
+    );
+    if (formulaType === 'host' || formulaType === 'host-id512') {
+      let storeFormulaIdentifier = `pet-store`;
+      if (formulaType === 'host-id512') {
+        storeFormulaIdentifier = `pet-store-id512:${formulaNumber}`;
+      }
+      // eslint-disable-next-line no-use-before-define
+      const petStore = await provideValueForFormulaIdentifier(
+        storeFormulaIdentifier,
+      );
+      const result = await E(petStore).identify(namePath);
+      // Use result if present, otherwise fallback to identifying from the host.
+      if (result !== undefined) {
+        return result;
+      }
+    }
+    // eslint-disable-next-line no-use-before-define
+    const origin = await provideValueForFormulaIdentifier(
+      originFormulaIdentifier,
+    );
+    return E(origin).identify(namePath);
```

Comment:
Given this is async tail recursive and has to at least reïfy pet stores (an unavoidably async operation), I withdraw my hope that `identify` can even be locally synchronous or transactional.

Given that it can’t be synchronous, there’s no reason to bundle up `locate(identifier)~.lookup(path)`. They can be peanut and butter: two separate methods that are better together.